app/Services/Nodes/NodeAutoDeployService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Services\Nodes;
+
+use App\Http\Controllers\Admin\NodeAutoDeployController;
+use App\Models\Node;
+
+class NodeAutoDeployService
+{
+    /**
+     * NodeAutoDeployService constructor.
+     */
+    public function __construct(
+        private readonly NodeAutoDeployController $nodeAutoDeployController
+    ) {
+    }
+
+    public function handle(Node $node): ?string
+    {
+        $service = $this->nodeAutoDeployController->__invoke(request(), $node);
```

Comment:
No.

Reply:
Don't DI controllers. Should be a service

Reply:
I'v made [my own branch](https://github.com/pelican-dev/panel/compare/main...RMartinOscar:panel:feature/AutoDeploy) since i don't have write permission on this PR
I'v refactored, fixed duplicated keys and notifications not showing up, added docker compatibility aswell as foreign key so memo must be unique to each user's keys
![image](https://github.com/user-attachments/assets/1a729fda-a3f6-48b1-aa34-5d2c61742d3a)
![image](https://github.com/user-attachments/assets/2fea359e-8f51-445b-8da2-c439677d80e5)

Reply:
Go ahead and open a new one 😊